tests/tls_e2e/host/host.cpp

```diff
@@ -0,0 +1,401 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <limits.h>
+#include <openenclave/host.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/tests.h>
+#include <pthread.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "tls_e2e_u.h"
+
+#define SERVER_PORT "12345"
+#define SERVER_IP "127.0.0.1"
+
+#define POSITIVE_TEST false
+#define NEGATIVE_TEST true
+#define SKIP_RETURN_CODE 2
+
+// A fatal error occured, eg the chain is too long or the vrfy callback failed
+#define FATAL_TLS_HANDSHAKE_ERROR -0x3000
+
+typedef struct _tls_thread_context_config
+{
+    oe_enclave_t* enclave;
+    struct tls_control_args args;
+} tls_thread_context_config_t;
+
+typedef struct _tls_test_configs
+{
+    tls_thread_context_config_t server;
+    tls_thread_context_config_t client;
+} tls_test_configs_t;
+
+typedef struct _test_cases_config
+{
+    const char* name;
+    struct tls_control_args args;
+    bool negative_test;
+} test_cases_config_t;
+
+typedef enum test_target
+{
+    server_target = 0,
+    client_target,
+    max_test_target_count
+} test_target_t;
+
+typedef enum test_config_type
+{
+    config_type_no_fault_injection = 0,
+    config_type_fail_oe_verify_tls_cert_scenario,
+    config_type_fail_cert_verify_callback_scenario,
+    config_type_fail_enclave_identity_verifier_callback_scenario,
+    max_test_config_type_count
+} test_config_type_t;
+
+oe_enclave_t* g_server_enclave = NULL;
+oe_enclave_t* g_client_enclave = NULL;
+int g_server_thread_exit_code = 0;
+int g_client_thread_exit_code = 0;
+pthread_mutex_t server_mutex;
+pthread_cond_t server_cond;
+bool g_server_condition = false;
+pthread_t server_thread_id;
+
+int server_is_ready()
+{
+    OE_TRACE_INFO("TLS server_is_ready!\n");
+    pthread_mutex_lock(&server_mutex);
+    g_server_condition = true;
+    pthread_cond_signal(&server_cond);
+    pthread_mutex_unlock(&server_mutex);
+    return 1;
+}
+
+oe_result_t enclave_identity_verifier(oe_identity_t* identity, void* arg)
+{
+    oe_result_t result = OE_VERIFY_FAILED;
+
+    (void)arg;
+    OE_TRACE_INFO("enclave_identity_verifier is called with parsed report:\n");
+
+    // Check the enclave's security version
+    OE_TRACE_INFO(
+        "identity.security_version = %d\n", identity->security_version);
+    if (identity->security_version < 1)
+    {
+        OE_TRACE_ERROR(
+            "identity.security_version check failed (%d)\n",
+            identity->security_version);
+        goto done;
+    }
+
+    // the unique ID for the enclave
+    // For SGX enclaves, this is the MRENCLAVE value
+    OE_TRACE_INFO("identity->signer_id :\n");
+    for (int i = 0; i < OE_UNIQUE_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
```

Comment:
Error: Replace signer_id to unique_id in lines 100 and 102

Reply:
good catch! Fixed